kloppy/infra/serializers/event/metrica/json_serializer.py

```diff
@@ -0,0 +1,311 @@
+                options : dict
+                    Options for deserialization of the Metrica Sports event json file. 
+                    Possible options are `event_types` (list of event types) to specify 
+                    the event types that should be returned. Valid types: "shot", "pass", 
+                    "carry", "take_on" and "generic". Generic is everything other than 
+                    the first 4. Those events are barely parsed. This type of event can 
+                    be used to do the parsing yourself.
+                    Every event has a 'raw_event' attribute which contains the original
+                    dictionary.
+                Returns
+                -------
+                dataset : EventDataset
+                Raises
+                ------
+
+                See Also
+                --------
+
+                Examples
+                --------
+                >>> serializer = MetricaEventsJsonSerializer()
+                >>> with open("events.json", "rb") as event_data, \
+                >>>      open("metadata.xml", "rb") as metadata:
+                >>>
+                >>>     dataset = serializer.deserialize(
+                >>>         inputs={
+                >>>             'event_data': event_data,
+                >>>             'metadata': metadata
+                >>>         },
+                >>>         options={
+                >>>             'event_types': ["pass", "take_on", "carry", "shot"]
+                >>>         }
+                >>>     )
+                """
+        self.__validate_inputs(inputs)
+        if not options:
+            options = {}
+
+        with performance_logging("load data", logger=logger):
+            raw_events = json.load(inputs["event_data"])
+            metadata = load_metadata(
+                inputs["metadata"], provider=Provider.METRICA
+            )
+
+        with performance_logging("parse data", logger=logger):
+
+            wanted_event_types = [
+                EventType[event_type.upper()]
+                for event_type in options.get("event_types", [])
+            ]
+
+            events = []
+            for raw_event in raw_events["data"]:
+                if raw_event["team"]["id"] == metadata.teams[0].team_id:
+                    team = metadata.teams[0]
+                elif raw_event["team"]["id"] == metadata.teams[1].team_id:
+                    team = metadata.teams[1]
+                else:
+                    raise Exception(
+                        f"Unknown team_id {raw_event['team']['id']}"
+                    )
+
+                player = team.get_player_by_id(raw_event["from"]["id"])
+                event_type = raw_event["type"]["id"]
+                subtypes = _parse_subtypes(raw_event)
+                period = [
+                    period
+                    for period in metadata.periods
+                    if period.id == raw_event["period"]
+                ][0]
```

Comment:
@koenvo Is there a better way to do this than indexing zero at the end of the comprehension?

Reply:
Might feed it to https://github.com/PySport/kloppy/pull/44#pullrequestreview-459658584 . But when it’s not found it should raise an exception